internal/schedules/filter.go

```diff
@@ -37,6 +39,23 @@ func NoRouteHandler(krakens map[string]kraken.Kraken, publisher Publisher) gin.H
 				c.JSON(http.StatusNotFound, gin.H{"message": "at least one filter is required"})
 				return
 			}
+			direction_type := c.QueryArray("direction_type")
+			if len(direction_type) > 0 {
+				dt := direction_type[0]
+				directions := []string{"backward", "forward", "all"}
+				if !utils.Contains(directions, dt) {
+					c.JSON(http.StatusBadRequest,
+						gin.H{"message": fmt.Sprintf("wrong direction type parameter selected : %s, it should be [forward, backward, all]]", dt)})
```

Comment:
You can directly use the variable `directions` for display (see [example](https://go.dev/play/p/WVL_3VdMNAm)).
```suggestion
						gin.H{"message": fmt.Sprintf("wrong direction type parameter selected : %s, it should be %v]", dt, directions)})
```

Also, you might want to split this huge line by extracting the `fmt.Sprintf(...)` into a variable.

Reply:
:+1: